go-controller/pkg/ovn/controller/services/node_tracker.go

```diff
@@ -310,3 +293,31 @@ func (nt *nodeTracker) getZoneNodes() []nodeInfo {
 	sort.SliceStable(out, func(i, j int) bool { return out[i].name < out[j].name })
 	return out
 }
+
+// isLocalZoneNode determines if a node is in our zone or not
+// This method is safe to call without a lock on nodeTracker as only the nt.zone
+// is read which is not changed after nodeTracker create.
+func (nt *nodeTracker) isLocalZoneNode(zone string, migrated bool) bool {
+	/** HACK BEGIN **/
+	// TODO(tssurya): Remove this HACK a few months from now. This has been added only to
+	// minimize disruption for upgrades when moving to interconnect=true.
+	// We want the legacy ovnkube-master to wait for remote ovnkube-node to
+	// signal it using "k8s.ovn.org/remote-zone-migrated" annotation before
+	// considering a node as remote when we upgrade from "global" (1 zone IC)
+	// zone to multi-zone. This is so that network disruption for the existing workloads
+	// is negligible and until the point where ovnkube-node flips the switch to connect
+	// to the new SBDB, it would continue talking to the legacy RAFT ovnkube-sbdb to ensure
+	// OVN/OVS flows are intact. Legacy ovnkube-master must not delete the service load
+	// balancers for this node till it has finished migration
+	if nt.zone == types.OvnDefaultZone {
+		if !migrated {
+			return true
```

Comment:
hmm I think you changed the functionality here a bit; if zone is global we want to return whatever is the value of migrated at this point in time, so `return migrated`
Basically keep same logic as: 
```
// isLocalZoneNode returns true if the node is part of the local zone.
func (bnc *BaseNetworkController) isLocalZoneNode(node *kapi.Node) bool {
	/** HACK BEGIN **/
	// TODO(tssurya): Remove this HACK a few months from now. This has been added only to
	// minimize disruption for upgrades when moving to interconnect=true.
	// We want the legacy ovnkube-master to wait for remote ovnkube-node to
	// signal it using "k8s.ovn.org/remote-zone-migrated" annotation before
	// considering a node as remote when we upgrade from "global" (1 zone IC)
	// zone to multi-zone. This is so that network disruption for the existing workloads
	// is negligible and until the point where ovnkube-node flips the switch to connect
	// to the new SBDB, it would continue talking to the legacy RAFT ovnkube-sbdb to ensure
	// OVN/OVS flows are intact.
	if bnc.zone == types.OvnDefaultZone {
		return !util.HasNodeMigratedZone(node)
	}
	/** HACK END **/
	return util.GetNodeZone(node) == bnc.zone
}
```